### DISCUSSION TITLE
Plugins use or not?

### DISCUSSION THREAD

**Discussion by tasos-ana** [ColorlibHQ/AdminLTE#4399] [Q&A]
Hello,

I am using on my app the plugin chartJS but the version of admin lte plugin is very old. Can i remove plugin and use the latest npm package? I lose something? Customization or something else? Can i install latest bootrstrap version instead of boostrap 4.6 version that included on plugin?

I want to known if plugins are customized/updated based on Admin lte or nothing changed? 

Thanks in advanced

**Comment by lenamtl** [comment]
Hi,

You can always use any plugins you want as long as it is compatible with your setup.

Bootstrap 5 is different from v4 see migration page https://getbootstrap.com/docs/5.0/migration/

The new AdminLTE template with Bootstrap 5 is not ready yet...

About plugin replacement you can check
https://github.com/ColorlibHQ/AdminLTE/discussions/3732

**Reply by tasos-ana** [reply]
Yes, but i must use admin lte plugins or i can install the original? For example chartjs on admin lte it's the version 2.9.3. can i install the latest version and keep it compatible with admin lte?

**Comment by lenamtl** [comment]
Hi,

AdminLte is just a template, you are free to use any Javascipt script and adapt the template CSS if required.

See the chartjs migration guide (it has **breaking changes**) https://www.chartjs.org/docs/latest/getting-started/v3-migration.html
From there you will see if your actual code required some adjustment, once your code is adjusted then test it on your dev server.
Check browser console to see if any error (Chrome right-click / inspect). Once it tested you can udate your production installation.

This apply to any JS script upgrade, some JS script may have dependency like jQuery so make sure to update dependency and test the whole script and the whole application, this may also required Bootstrap update for example to use jQuery 3.60 you need to use BS3 v3.3.7 and later.

I'm still using the old AdminLte v2 BS3 and now I do upgrade the JS script librairies to the more recent version for security reason and to benefit new features. I'm waiting AdminLte v4 to update to Bootstrap 5...
I don't think it worth to update to Bootstrap 4 has BS5 is stable since a year and there are always breaking changes between version especially BS5 as they drop jQuery.
https://getbootstrap.com/docs/5.0/migration/#javascript
